lib/authentication/service/auth_provider.dart

```diff
@@ -184,6 +148,38 @@ class AuthProvider with ChangeNotifier {
     return _firebaseUser.uid;
   }
 
+  Future<void> _emigrate(Map<String,dynamic> data) async{
+    if(data['class'] != null && data['class'] is Map){
```

Comment:
I'd make this check outside of this method and just `assert` it here. As in, only call the method if it makes sense to (essentially, `if (dataHasOldFormat) convertToNewFormat()`).